src/components/Form/Form.tsx

```diff
@@ -1,21 +1,13 @@
-import { Dispatch, SetStateAction, useState } from "react";
-import { EFormData } from "../../enums/Form";
-import { FormActions } from "../../reducers/form-reducer";
-import { FormData, FormErrors } from "../../types/FormData";
+import { validateForm } from "../../helpers/helpers";
+import { useForm } from "../../hooks/useForm";
 import Dates from "../Dates/Dates";
 import CardHolderNameInput from "./components/CardHolderNameInput";
 import CardNumberInput from "./components/CardNumberInput";
 import CVCInput from "./components/CVCInput";
-import { validateForm } from "../../helpers/helpers";
 
-type FormProps = {
-  state: FormData;
-  dispatch: Dispatch<FormActions>;
-  setIsSubmitted: Dispatch<SetStateAction<boolean>>;
-};
+const Form = () => {
 
-const Form = ({ state, dispatch, setIsSubmitted }: FormProps) => {
-  const [errors, setErrors] = useState<FormErrors>({});
+  const { state, setErrors, setIsSubmitted } = useForm()
 
   const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
```

Comment:
this handleChange should be ideal to have it in the context so you remove most of the logic to the context and keep clean the component